src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -259,8 +260,17 @@ namespace cryptonote
 
     m_fakechain = test_options != NULL;
     bool r = handle_command_line(vm);
+    bool testnet = command_line::get_arg(vm, command_line::arg_testnet_on);
+    auto p2p_bind_arg = testnet ? nodetool::arg_testnet_p2p_bind_port : nodetool::arg_p2p_bind_port;
+    std::string m_port = command_line::get_arg(vm, p2p_bind_arg);
+    std::string m_config_folder_mempool = m_config_folder;
+
+    if ((!testnet && m_port != std::to_string(::config::P2P_DEFAULT_PORT))
+        || (testnet && m_port != std::to_string(::config::testnet::P2P_DEFAULT_PORT))) {
+      m_config_folder_mempool = m_config_folder_mempool + "/" + m_port;
```

Comment:
On other parts of the system boost::filesystem::path is used as you denote, but in those two cases std::string is used.

In the link from my previous comment and in the following one slashes are already used, so behaviour should not change.

https://github.com/monero-project/monero/blob/master/src/cryptonote_core/tx_pool.cpp#L714